### PR TITLE
[Early Access] Gary Centcoms new hit game: Space station: battle royale. The all new free, online third person robust-them up complete with the famous robusting rocket, the place where dreams are made. Preorder now on the epic flames store for a free 1 month battle pass.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Improves battle royale a bunch.
 - Battle royale is now on a new map loaded when BR is started.
 - Battle royale now has a shuttle you can jump off at any time.
Its pretty cool
 - Loot pool is updated to include more guns
 - Guns do less damage.
 - Players start with a harvesting tool capable of quickly destroying objects.
 - Upon death players items will be thrown around

image

Video was too big so here is a single screenshot of my gameplay.

image

image

## Why It's Good For The Game

Its really good game, 10/10.

## Changelog
:cl: Gary Centcom CEO of Epic Flames
add: Battle royale map
add: Battle royale robusting rocket.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
early access btw.
no bugs only features.

**Comment by wraith-54321** [comment]
yeah, battle royale is really cool but its very rarely used

**Comment by eeSPee** [comment]
Probably will never see the light of day because it doesn't comment out a feature everyone likes.

**Comment by PowerfulBacon** [comment]
ok seems good to me

File: code/modules/battle_royale/battle_royale_species.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/datum/species/human/battleroyale
+	name = "Battle Royale Soldier" //inherited from the real species, for health scanners and things
+	id = "battleroyale"
+	limbs_id = "human"
+	species_traits = list(NOTRANSSTING) //all of these + whatever we inherit from the real species
+	inherent_traits = list(TRAIT_NOHUNGER,TRAIT_PIERCEIMMUNE,TRAIT_NODISMEMBER,TRAIT_VIRUSIMMUNE,TRAIT_NODISMEMBER,TRAIT_ALWAYS_CLEAN,TRAIT_FEARLESS)
```
Comment (WhyIsCaeciliusTaken) [inline code/modules/battle_royale/battle_royale_species.dm]: TRAIT_PIERCEIMMUNE is listed twice

File: code/modules/battle_royale/battle_royale_drops.dm [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	/obj/item/gun/ballistic/shotgun = 5,
+	/obj/item/gun/ballistic/shotgun/automatic/combat = 1,
+	/obj/item/gun/ballistic/automatic/ar = 2,
+	/obj/item/gun/ballistic/rifle/boltaction = 4,
+	/obj/item/pneumatic_cannon/speargun = 1,
+	/obj/item/gun/ballistic/automatic/mini_uzi = 5,
+	/obj/item/gun/energy/lasercannon = 1,
+	/obj/item/gun/energy/kinetic_accelerator/crossbow/large = 4,
+	/obj/item/gun/energy/e_gun/nuclear = 1,
+	/obj/item/gun/ballistic/automatic/proto = 1,
```
Comment (WhyIsCaeciliusTaken) [inline code/modules/battle_royale/battle_royale_drops.dm]: This SMG spawns with no firing pin. You need the unrestricted subtype.

File: code/modules/battle_royale/battle_royale_drops.dm [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	/obj/item/gun/ballistic/automatic/gyropistol = 2,
+	/obj/item/gun/ballistic/automatic/pistol/suppressed = 6,
+	/obj/item/gun/ballistic/shotgun/doublebarrel = 4,
+	/obj/item/gun/ballistic/shotgun = 5,
+	/obj/item/gun/ballistic/shotgun/automatic/combat = 1,
+	/obj/item/gun/ballistic/automatic/ar = 2,
+	/obj/item/gun/ballistic/rifle/boltaction = 4,
+	/obj/item/pneumatic_cannon/speargun = 1,
+	/obj/item/gun/ballistic/automatic/mini_uzi = 5,
+	/obj/item/gun/energy/lasercannon = 1,
```
Comment (WhyIsCaeciliusTaken) [inline code/modules/battle_royale/battle_royale_drops.dm]: No firing pin.

File: code/modules/battle_royale/battle_royale_drops.dm [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+	/obj/item/storage/box/teargas = 6,
+	/obj/item/storage/toolbox/syndicate = 8,
+	/obj/item/storage/box/syndie_kit/bee_grenades = 1,
+	/obj/item/storage/box/syndie_kit/centcom_costume = 1,
+	/obj/item/storage/box/syndie_kit/chameleon = 1,
+	/obj/item/storage/box/syndie_kit/chemical = 1,
+	/obj/item/storage/box/syndie_kit/emp = 1,
+	/obj/item/storage/box/syndie_kit/imp_adrenal = 1,
+	/obj/item/storage/box/syndie_kit/imp_freedom = 1,
+	/obj/item/storage/box/syndie_kit/imp_radio = 1,
```
Comment (WhyIsCaeciliusTaken) [inline code/modules/battle_royale/battle_royale_drops.dm]: What's the point of a radio implant in a battle royale?

**Review by WhyIsCaeciliusTaken — CHANGES_REQUESTED** [review]

File: code/modules/battle_royale/battle_royale_species.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/battle_royale/battle_royale_species.dm]: its so nice I had to put it twice

File: code/modules/battle_royale/battle_royale_drops.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/modules/battle_royale/battle_royale_drops.dm]: fixed

File: code/modules/battle_royale/battle_royale_species.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/battle_royale/battle_royale_species.dm]: wait I only see it once

Comment (PowerfulBacon) [reply code/modules/battle_royale/battle_royale_species.dm]: oh TRAIT_NODISMEMBER is twice

**Comment by PowerfulBacon** [comment]
Needs more guns, harvesting tool seems to not work

**Comment by PowerfulBacon** [comment]
[19:51:04] Runtime in projectile.dm, line 148: undefined variable /datum/species/human/battleroyale/var/pixel_x

**Comment by PowerfulBacon** [comment]
this was actually a lot of fun and a big improvement

**Pull request closed without merging**